server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageBlobsDAO.java

```diff
@@ -148,8 +152,14 @@ public InputStream read(BucketName bucketName, BlobId blobId) throws ObjectStore
 
     }
 
-    public void deleteContainer() {
-        blobStore.deleteContainer(containerName.value());
+    @Override
+    public BucketName getDefaultBucketName() {
+        return defaultBucketName;
```

Comment:
I really would love tests for `getDefaultBucketName()` in a separated commit given various configurations:

 - What is the default bucket name with default empty configuration (it SHOULD be "default")
 - What is the default bucket when the namespace option is specified (it SHOULD be namespace without any alteration)
 - What the default bucket is when only a prefix is specified (it should be "prefix-default")
 - Is it valid to specify a namespace AND a prefix (answer: no, as the prefix will not apply to the namespace)

Also, when there is a prefix "prefix_" configured, and that I create a bucket named "custom" we need to ensure that the underlying bucket being used is **prefix_custom**...